### PR TITLE
Upgrade to React Router v6 functions

### PR DESCRIPTION
**Pull request by ManpreetSL**
React Router DOM package is changing things like `useHistory()` in v5 to `useNavigate()` in v6. See [here](https://reactrouter.com/en/main/upgrading/v5) for more details.
The app currently uses the old `useHistory()` hook from the React Router package, but the app has v6 installed, so we'd like to update usages of the old function wherever possible.
We also want to replace `<Redirect />` with `<Navigate />`.

This PR:
* Replaces `useHistory()` usages with `useNavigate()`, the new function for v6 of the router package.
* Replaces `<Redirect />` with `<Navigate />`.

Implements #678.

### CODE REVIEW

**Review by saihaj — APPROVED** [review]
thanks for refactoring this!

File: apps/frontend/src/Controller/index.tsx [hunk 1]
```diff
@@ -249,18 +249,18 @@ const Controller = ( props ) => {
...
                 onHover={setHovered}
                 renderContent={() => BarComponent && (
-                <BarComponent
-                  {...props}
-                  onHover={setHovered}
-                />
+                  <BarComponent
+                    {...props}
+                    onHover={setHovered}
+                  />
```
[the remark this answers is not in the document]
Comment (ManpreetSL) [reply apps/frontend/src/Controller/index.tsx:255]: I think my VS code is set to use ESLint as the formatter. Is it a problem?

[the remark this answers is not in the document]
Comment (saihaj) [reply apps/frontend/src/Controller/index.tsx:255]: I think we have our ESLint setup not sure if this rule was not picked by that

**Review by Harjot1Singh — APPROVED** [review]

File: apps/frontend/src/Controller/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ManpreetSL) [reply apps/frontend/src/Controller/index.tsx:255]: Do I need to change anything?

**Pull request merged**